hashes/src/sha512_256.rs

```diff
@@ -0,0 +1,262 @@
+// Bitcoin Hashes Library
+// Written in 2018 by
+//   Andrew Poelstra <apoelstra@wpsoftware.net>
+//
+// To the extent possible under law, the author(s) have dedicated all
+// copyright and related and neighboring rights to this software to
+// the public domain worldwide. This software is distributed without
+// any warranty.
+//
+// You should have received a copy of the CC0 Public Domain Dedication
+// along with this software.
+// If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+//
+
+// This module is largely copied from the rust-crypto ripemd.rs file;
+// while rust-crypto is licensed under Apache, that file specifically
+// was written entirely by Andrew Poelstra, who is re-licensing its
+// contents here as CC0.
+
+//! SHA512_256 implementation.
+//!
+
+use core::{cmp, str};
+use core::ops::Index;
+use core::slice::SliceIndex;
+
+use crate::{Error, sha512, HashEngine as _, hex};
+
+const BLOCK_SIZE: usize = 128;
+
+/// Engine to compute SHA512/256 hash function.
+#[derive(Clone)]
```

Comment:
I would appreciate a comment pointing out this is neither SHA256 nor SHA512 and some brief explanation (link) of the hash function.

Reply:
Added a brief explanation and the link to the original paper.

Reply:
Perfect!